src/app/components/formElements/Slider/SliderInput.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { FocusEvent, useEffect, useRef } from 'react';
+import styles from './slider.module.css';
+import { DefaultValue, SliderInputProps } from '@/app/interfaces/FormElements';
+import { FieldValues } from 'react-hook-form';
+import Label from '../Label/Label';
+import HelperText from '../HelperText/HelperText';
+import ErrorBox from '../ErrorBox/ErrorBox';
+import useDefaultValue from '@/app/hooks/useDefaultValue';
+
+const SliderInput = <TFieldValues extends FieldValues>({
+  name,
+  label,
+  showLabel = false,
+  helperText,
+  defaultValue = 50 as DefaultValue<TFieldValues>,
+  onChange,
+  defaultErrorMessage,
+  renderProps,
+  methods,
+}: SliderInputProps<TFieldValues>) => {
+  const {
+    field,
+    formState: { errors },
+  } = renderProps;
+  const error = errors[name];
+  const { disabled, onBlur, value } = field;
+
+  const sliderRef = useRef<HTMLInputElement>(null);
+
+  useDefaultValue<TFieldValues>({
+    renderProps,
+    defaultValue,
+    setValue: methods.setValue,
+  });
+
+  const setValue = (value: number) => {
+    field.onChange(value);
+    onChange?.(value);
+    if (sliderRef.current) {
+      sliderRef.current.style.setProperty('--slider-value', `${value}%`);
+    }
```

Comment:
When you call `field.onChange`, it enters the SliderInput. You can just use the current value to determine the css variable.